storage/aptosdb/src/lib.rs

```diff
@@ -593,27 +593,33 @@ impl AptosDB {
         txns_to_commit: &[TransactionToCommit],
         first_version: u64,
         cs: &mut ChangeSet,
-    ) -> Result<HashValue> {
+    ) -> Result<(HashValue, Option<Version>)> {
         let last_version = first_version + txns_to_commit.len() as u64 - 1;
 
         // Account state updates. Gather account state root hashes
-        {
+        let updated_state_version = {
             let _timer = OTHER_TIMERS_SECONDS
                 .with_label_values(&["save_transactions_state"])
                 .start_timer();
 
-            let account_state_sets = txns_to_commit
+            let state_updates_vec = txns_to_commit
                 .iter()
                 .map(|txn_to_commit| txn_to_commit.state_updates())
                 .collect::<Vec<_>>();
+            let updated_state_version = state_updates_vec
```

Comment:
This looks confusing and hacky to me - is the intention to find the version just before the state checkpoint? If so, can we search for the `StateCheckPoint` txn in `txns_to_commit` instead?